aten/src/ATen/core/boxing/impl/make_boxed_from_unboxed_functor.h

```diff
@@ -42,21 +42,18 @@ using supported_primitive_arg_types = guts::typelist::typelist<
     at::Tensor,
     at::Scalar,
     c10::QScheme,
-    c10::ScalarType>;
-
-template <class T, bool AllowDeprecatedTypes, class Enable = void>
-struct assert_is_valid_input_type {
-  assert_is_valid_input_type() {
-    auto tmap = c10::getCustomClassTypeMap();
-    TORCH_CHECK(
-        c10::isCustomClassRegistered<T>(),
-        "Tried to use undefined class as input argument");
-  }
-  };
-
-  template<class T, bool AllowDeprecatedTypes>
-  struct assert_is_valid_input_type<T, AllowDeprecatedTypes, std::enable_if_t<guts::typelist::contains<supported_primitive_arg_types, T>::value>> {
-    // everything is ok, this is a primitive type
+    c10::ScalarType
+  >;
+
+  template<class T, bool AllowDeprecatedTypes, class Enable = void> struct assert_is_valid_input_type {
+    assert_is_valid_input_type() {
+      guts::if_constexpr<guts::typelist::contains<supported_primitive_arg_types, T>::value>([] {
```

Comment:
personally i think it wouldn't hurt to smack a comment above the `if_constexpr`s, that briefly describes the compile-time computation that's going on